### PR TITLE
Fix: not set ecmaVersion to 6 when sourceType=module (fixes #9687)

### LINKED ISSUE #9687: False negatives with sourceType: module and ecmaVersion <= 5
**Tell us about your environment**

* **ESLint Version:**
v4.12.1
* **Node Version:**
8.4.0
* **npm Version:**
5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    "parserOptions": {
        "sourceType": "module",
        "ecmaVersion": 5,
    }
}
```

</details>

Running against this file, ecmaVersion is ignored (could also be '3'):
 
```js
// es6.js
const x = () => `hi`;
```

**What did you expect to happen?**

Either, I would see an error like the below, rejecting an unsupported config:

```bash
> eslint -c withoutModule.config.js es6.js
Incompatible config: sourceType: module cannot be used with ecmaVersion <= 5
```

or the (admittedly niche*) use-case of ES6 modules with ES5 syntax would be supported, and I'd see errors:

```bash
> eslint -c withoutModule.config.js es6.js
es6.js
  1:1  error  Parsing error: The keyword 'const' is reserved

✖ 1 problem (1 error, 0 warnings)
```

which I get correctly with this config:

```js
// withoutModule.config.js
module.exports = {
    "parserOptions": {
        "ecmaVersion": 5,
    }
}
```

*niche: I was using rollup with some old ES5 code, and only wanted to modify the code to add imports. I tried the above, and thought "cool, it's working", but actually it was silently ignoring the ecmaVersion.

**What actually happened? Please include the actual, raw output from ESLint.**

No errors are reported:

```bash
> eslint es6.js
> echo $?
0
```

### PR DESCRIPTION
**Pull request by aladdin-add**
PR #11610 was aimed to fix the issue #9687, however it did not work,
since ecmaVersion has been normalized to 6 (when sourceType is setting to 'module').

this PR removed the normalize behaviour, and fixed failing tests.

**What is the purpose of this pull request? (put an "X" next to item)**

[x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))

**What changes did you make? (Give an overview)**

**Is there anything you'd like reviewers to focus on?**
no.

### CODE REVIEW

File: tests/lib/linter.js [hunk 1]
```diff
@@ -4422,25 +4422,41 @@ describe("Linter", () => {
...
 
         it("should properly parse default export statements when sourceType is module", () => {
             const code = "export default function initialize() {}";
-            const messages = linter.verify(code, { parserOptions: { sourceType: "module" } });
+            const messages = linter.verify(code, { parserOptions: { ecmaVersion: 6, sourceType: "module" } });
 
             assert.strictEqual(messages.length, 0);
         });
 
+        // https://github.com/eslint/eslint/issues/9687
```
Comment (aladdin-add) [inline tests/lib/linter.js]: this is new added tests.

File: tests/lib/linter.js [hunk 2]
```diff
@@ -4422,25 +4422,41 @@ describe("Linter", () => {
...
+            assert.deepStrictEqual(messages.length, 1);
+            assert.ok(/Invalid ecmaVersion/u.test(messages[0].message));
+
+            messages = linter.verify("", { parserOptions: { sourceType: "foo" } });
+            assert.deepStrictEqual(messages.length, 1);
+            assert.ok(/Invalid sourceType/u.test(messages[0].message));
+
+            messages = linter.verify("", { parserOptions: { ecmaVersion: 5, sourceType: "module" } });
+            assert.deepStrictEqual(messages.length, 1);
+            assert.ok(/sourceType 'module' is not supported when ecmaVersion < 2015/u.test(messages[0].message));
```
Comment (g-plane) [inline tests/lib/linter.js]: Can we use `String.prorotype.includes` instead of very long regexp here?

File: lib/linter.js [hunk 3]
```diff
@@ -280,30 +280,6 @@ function getDirectiveComments(filename, ast, ruleMapper) {
...
-    if (isModule && (!ecmaVersion || ecmaVersion < 6)) {
-        return 6;
-    }
-
-    /*
-     * Calculate ECMAScript edition number from official year version starting with
-     * ES2015, which corresponds with ES6 (or a difference of 2009).
-     */
-    if (ecmaVersion >= 2015) {
-        return ecmaVersion - 2009;
```
Comment (mysticatea) [inline lib/linter.js]: Removal of this normalization may be an unintentional breaking change for custom parsers and scope managers. At least, `eslint-scope` doesn't support the year form.

Comment (aladdin-add) [reply lib/linter.js]: good catch! 

it may be a breaking, will add it back. Though I'm sure we should normalize a 3rd-party parser option.

**Comment by platinumazure** [comment]
Should this be considered a breaking change?

**Comment by aladdin-add** [comment]
Or it can be a bugfix -- the breaking change should have been introduced in #11610 . I fancy either is fine.

**Comment by not-an-aardvark** [comment]
It looks like there are merge conflicts.

**Comment by aladdin-add** [comment]
will update it later.

**Pull request merged**